packages/core/core/src/TransformerRunner.js

```diff
@@ -229,6 +232,17 @@ async function finalize(asset: Asset, generate: GenerateFunc): Promise<Asset> {
   return asset;
 }
 
+async function checkPipelineCacheInvalidation(
+  pipeline: Array<Transformer>,
+  cacheEntry: CacheEntry
+): boolean {
+  return pipeline.every(
+    transformer =>
+      !transformer.shouldInvalidateCache ||
+      !transformer.shouldInvalidateCache(cacheEntry.meta)
```

Comment:
The `meta` property would be on an individual asset rather than the whole cache entry. The cache entry can potentially hold multiple assets. For example, in a Vue file, there is typically a JS section, a template, and some CSS. This is the output of a single transformer for one file, so it is stored in one cache entry for that file.

We'll need to do this check for each asset in the cache entry.